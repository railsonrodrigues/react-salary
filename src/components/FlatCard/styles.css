.flatCard {
  width: 100%;
  padding: 15px 30px;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  box-shadow: 0 0 10px 0 rgba(0,0,0,0.3);
}