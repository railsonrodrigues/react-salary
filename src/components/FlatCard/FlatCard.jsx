import React from 'react';
import './styles.css';

export default function FlatCard ({ title, value, percentage }) {
  return (
    <div className="card flatCard">
      <strong>{ title }</strong>
      <span>{ value }</span>
      <span>{ percentage }</span>
    </div>
  )
}