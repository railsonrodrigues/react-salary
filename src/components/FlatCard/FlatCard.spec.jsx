import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom'
import React from 'react';
import FlatCard from './FlatCard';

describe('<FlatCard />', () => {
  it('should render FlatCard correctly', () => {
    const {container} = render(<FlatCard title="INSS" value="700" percentage="30" />)
    const flatCard = container.querySelector('div');
    const titleElement = container.querySelector('h3.sc-dlnjwi');
    const amountElement = container.querySelector('span.sc-gtsrHT');
    const percentageElement = container.querySelector('span.kquQli');

    expect(flatCard).toBeInTheDocument();
    expect(titleElement).toBeInTheDocument();
    expect(amountElement).toBeInTheDocument();
    expect(percentageElement).toBeInTheDocument();
  })
  it('should match snapshot', () => {
    const {container} = render(<FlatCard title="INSS" value="700" percentage="30" />)
    expect(container.firstChild).toMatchSnapshot();
  })
})