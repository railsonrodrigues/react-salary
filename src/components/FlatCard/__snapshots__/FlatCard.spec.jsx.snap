// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FlatCard /> should match snapshot 1`] = `
<div
  class="sc-bdnxRM jPHcVx"
>
  <h3
    class="sc-dlnjwi ctIjkz"
  >
    INSS
  </h3>
  <span
    class="sc-gtsrHT YfnJr"
  >
    700
  </span>
  <span
    class="sc-gtsrHT kquQli"
  >
    30
  </span>
</div>
`;
